src/domain/flow-system/services/src/messages/flow_message_consumer_mediator.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use dill::{component, interface, Catalog};
+use internal_error::InternalError;
+use kamu_flow_system::{FlowConfigurationUpdatedMessage, FlowServiceUpdatedMessage};
+use messaging_outbox::{consume_deserialized_message, ConsumerFilter, MessageConsumersMediator};
+
+use super::MESSAGE_PRODUCER_KAMU_FLOW_SERVICE;
+use crate::MESSAGE_PRODUCER_KAMU_FLOW_CONFIGURATION_SERVICE;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[component(pub)]
+#[interface(dyn MessageConsumersMediator)]
+pub struct FlowMessageConsumerMediator {}
+
+#[async_trait::async_trait]
+impl MessageConsumersMediator for FlowMessageConsumerMediator {
+    fn get_producers(&self) -> &[&'static str] {
+        &[
+            MESSAGE_PRODUCER_KAMU_FLOW_CONFIGURATION_SERVICE,
+            MESSAGE_PRODUCER_KAMU_FLOW_SERVICE,
+        ]
+    }
+
+    async fn consume_message<'a>(
+        &self,
+        catalog: &Catalog,
+        producer_name: &str,
+        consumer_filter: ConsumerFilter<'a>,
+        content_json: serde_json::Value,
+    ) -> Result<(), InternalError> {
+        match producer_name {
+            MESSAGE_PRODUCER_KAMU_FLOW_CONFIGURATION_SERVICE => {
```

Comment:
I want to get rid of mediators so bad. Their logic is so repetitive now ... there has to be a way to generalize it.

Reply:
I'm imagining something like this:

```rust
catalog_builder
  .add(outbox::Dispatcher::for::<TaskServiceMessage>(MESSAGE_PRODUCER_TASK_SERVICE))
  .add(outbox::Dispatcher::for::<FlowServiceMessage>(MESSAGE_PRODUCER_FLOW_SERVICE))
```

i.e. there is a generic implementation of a "dispatcher" that deserialized JSON into the right type and sends it off to the typed handlers.

And `Dispatcher::for` returns a builder with producer ID contained as metadata, so that outbox service could select the correct dispatcher implementation.

Reply:
Couldn't solve it with a builder, the solution is a bit more verbose, but still, mediators are gone.